File: docs/.validation/python/pre-tool-use_16.py
# Source: hooks/pre-tool-use.md:113
import asyncio

async def main():
    async def on_pre_tool_use(input_data, invocation):
        print(f"[{invocation['session_id']}] Calling {input_data['toolName']}")
        print(f"  Args: {input_data['toolArgs']}")
        return {"permissionDecision": "allow"}
    
    session = await client.create_session({
        "hooks": {"on_pre_tool_use": on_pre_tool_use}
    })

asyncio.run(main())

File: docs/.validation/csharp/pre-tool-use_16.cs
// Source: hooks/pre-tool-use.md:149
var session = await client.CreateSessionAsync(new SessionConfig
{
    Hooks = new SessionHooks
    {
        OnPreToolUse = (input, invocation) =>
        {
            Console.WriteLine($"[{invocation.SessionId}] Calling {input.ToolName}");
            Console.WriteLine($"  Args: {input.ToolArgs}");
            return Task.FromResult<PreToolUseHookOutput?>(
                new PreToolUseHookOutput { PermissionDecision = "allow" }
            );
        },
    },
});